classes/client/s3_client.php

```diff
@@ -80,11 +80,21 @@ public function __wakeup() {
     }
 
     public function set_client($config) {
-        $this->client = S3Client::factory(array(
-        'credentials' => array('key' => $config->s3_key, 'secret' => $config->s3_secret),
-        'region' => $config->s3_region,
-        'version' => AWS_API_VERSION
-        ));
+        $proxy = $this->construct_proxy_string($config);
```

Comment:
Hi @tuanngocnguyen  It's still a lot of code duplication. What do you think about abstracting the array we are passing to S3Client::factory() to a function? There we can check for proxy and add additional key if required...